src/components/Viewers/AudioPlayer.jsx

```diff
@@ -0,0 +1,150 @@
+import { useState, useEffect, useRef } from "react";
+
+import PauseCircleOutlineIcon from "@mui/icons-material/PauseCircleOutline";
+import PlayCircleOutlineIcon from "@mui/icons-material/PlayCircleOutline";
+import SkipNextIcon from "@mui/icons-material/SkipNext";
+import SkipPreviousIcon from "@mui/icons-material/SkipPrevious";
+import { Box, IconButton, Button } from "@mui/material";
+
+const AudioPlayer = (item) => {
+  const [currentAudioIndex, setCurrentAudioIndex] = useState(0); // Track current audio index
+  const [audio, setAudio] = useState(new Audio(item.content[0]));
+  const [isPlaying, setIsPlaying] = useState(false);
+  const [progress, setProgress] = useState(0); // Progress of the song in percentage
+  const progressBarRef = useRef(); // Reference to the progress bar for calculating click positions
+
+  useEffect(() => {
```

Comment:
it would be nice if move  useEffects  to custom hook  , more readable and clean code